src/fides/api/service/connectors/dynamic_erasure_email_connector.py

```diff
@@ -0,0 +1,342 @@
+            )
+            return
+
+        # Get the graph dataset from the dataset config
+        graph_dataset: GraphDataset = dataset_config.get_graph()
+
+        # Field should be of the form collection_name.field_name, potentially with nested fields
+        # i.e collection_name.field_name.nested_field_name
+        # so splitting by "." should give a list of length greater than or equal to 2.
+        split_field = dataset_reference.field.split(".")  # pylint: disable=no-member
+
+        if len(split_field) < 2:
+            logger.error(
+                "Invalid dataset reference field '{}' for dataset {}. Skipping erasure email send for connector: '{}'.",
+                dataset_reference.field,  # pylint: disable=no-member
+                dataset_key,
+                self.configuration.key,
+            )
+            return
+
+        # Build the CollectionAddress using the dataset name and the collection name
+        collection_name = split_field[0]
+        collection_address = CollectionAddress(graph_dataset.name, collection_name)
+        # And get the field name from the datasetreference
+        field_name = ".".join(split_field[1:])
+
+        # Get the corresponding ConnectionConfig instance
+        connection_config: ConnectionConfig = (
+            db.execute(
+                db.query(ConnectionConfig).filter(
+                    ConnectionConfig.id == dataset_config.connection_config_id
+                )
+            )
+            .scalars()
+            .first()
+        )
+
+        # Instatiate the ConnectionConfig's connector so that we can execute the query
+        # to retrieve the email addresses based on the custom request fields
+        connector = get_connector(connection_config)
+
+        skipped_privacy_requests: List[str] = []
+        # We'll batch identities for each email address
+        # so we'll only send 1 email to each email address
+        batched_identities: Dict[str, List[str]] = {}
+
+        for privacy_request in privacy_requests:
+            try:
+                email = self.get_email_address_from_custom_request_fields(
+                    connector,
+                    graph_dataset,
+                    privacy_request,
+                    collection_address,
+                    field_name,
+                )
+
+                user_identities: Dict[str, Any] = (
+                    privacy_request.get_cached_identity_data()
+                )
+                filtered_user_identities: Dict[str, Any] = (
+                    filter_user_identities_for_connector(self.config, user_identities)
+                )
+
+                # If we couldn't get the email or the user identities, we skip the privacy request
+                if not email or not filtered_user_identities:
+                    skipped_privacy_requests.append(privacy_request.id)
+                    self.add_skipped_log(db, privacy_request)
+                    continue
+
```

Comment:
same question here, do we skip these privacy requests or do we want it to error? The existing generic erasure email connector skips the request if `filtered_user_identities` is None

Reply:
maybe we break this up into two? It seems like a legitimate use case where the user identity doesn't exist so I'd say we skip it there.  But if the email doesn't exist, that seems like a problem, maybe it should fail. what do you think @galvana